### DISCUSSION TITLE
How to make full clone of strapi project?

### DISCUSSION THREAD

**Discussion by mshilenkov** [strapi/strapi#7642] [Legacy Topics]
Could anybody explain step-by-step how to make a full development version's clone of strapi project from a computer to another computer by GitHub for example? I need to have all previously created records and data including admin profile and roles. What files should I include in the commit? How .gitignore file should looks like etc? Thank you for any help.

**Comment by yifeiyin** [comment]
Not an expert here, but to my knowledge, the process can be different for different configurations. e.g. what database you are using, are you using media libraries and if so where are the files stored), etc.

All records, admin profile, permissions are stored in the database. So if you copy over the database, all these data are preserved. File are stored separately however, so if you have uploaded files to strapi you need to copy over that too. The location of these files are different depends on you configurations.

btw, committing the database to the repo is generally considered as a bad idea as any later changes to the database will be reflected in git; you probably want to do this manually instead.

**Reply by derrickmehaffy** [reply]
@yifeiyin is correct, users, roles, permissions, and content-manager views are all stored in the database. It's much safer to transfer these items outside of Git (especially users and permissions). We do have a dump/restore command for content-manager views, but it dumps the entire `core_store` table to json and can accidentally leak any auth provider secret keys so you have to use it with caution.

Docs for that can be found here: 
- https://strapi.io/documentation/v3.x/cli/CLI.html#strapi-configuration-dump-config-dump
- https://strapi.io/documentation/v3.x/cli/CLI.html#strapi-configuration-restore-config-restore

Users, Permissions, Roles will have to be dumped/restored manually

**Reply by mshilenkov** [reply]
Thank you very much for the answers.
